### PR TITLE
core - policy - add metadata key

### PR DESCRIPTION
**Pull request by JohnHillegass**
Proposed changes if https://github.com/cloud-custodian/cloud-custodian/issues/5968 is accepted

### CODE REVIEW

File: c7n/schema.py [hunk 1]
```diff
@@ -261,6 +261,7 @@ def generate(resource_types=()):
                 'comments': {'type': 'string'},
                 'description': {'type': 'string'},
                 'tags': {'type': 'array', 'items': {'type': 'string'}},
+                'metadata': {'type': 'object'},
```
Comment (kapilt) [inline c7n/schema.py:264]: this needs to be type restricted to ie string key value to basic types, not a wild card, see basic_dict in common definitions.

Comment (JohnHillegass) [reply c7n/schema.py]: Fair point, hadn't considered restricting it. This should likely ever only be a string so I set a reference to string_dict.

Comment (kapilt) [reply c7n/schema.py]: if your restricting it to strings its not any different then using "key:value" style labels

unless you have a particular rationale, this should likely support basic types.

Comment (JohnHillegass) [reply c7n/schema.py]: Looking at it fresh this morning, I agree that boolean and number are probably going to be useful for this too. Updated to use basic_dict.

**Pull request merged**